pkg/cmd/openshift-sdn/cmd.go

```diff
@@ -194,15 +196,20 @@ func (sdn *OpenShiftSDN) Init() error {
 
 // Start starts the network, proxy, and informers, then returns.
 func (sdn *OpenShiftSDN) Start(stopCh <-chan struct{}) error {
-	klog.Infof("Starting node networking (%s)", version.Get().String())
+	klog.Infof("Starting node service proxy (%s)", version.Get().String())
 
 	serviceability.StartProfiler()
+	sdn.runProxy()
+	sdn.informers.start(stopCh)
+
+	klog.Infof("Waiting for proxy rules to sync...")
+	<-sdn.proxyReadyChan
+
+	klog.Infof("Proxy rules successfully synced, starting node networking")
```

Comment:
/hold
Historically, it was necessary for the SDN to start before the proxy, but I don't remember why. (The explanation should exist in github somewhere.) It is possible (perhaps even likely) that whatever reason it was no longer applies, but I definitely wouldn't want to merge this now without figuring that out first.

Reply:
Yeah, I was looking in to that and couldn't find anything. The one dependency I did find was that kube-proxy needs KUBE-MARK-MASQ to be created (which is done by the kubelet). Maybe back in the days of the monolithic node process that was an issue.